### PR TITLE
Internal: Create repository for `global-classes` [EDS-572]

### PR DESCRIPTION
**Pull request by Omerisra6**

### CODE REVIEW

File: modules/global-classes/repository.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			throw new \Exception( "Global class with id ${id} not found" );
+		}
+
+		$this->kit->update_json_meta( self::META_KEY, [
+			'data' => $all->get_data()->except( [ $id ] )->all(),
+			'order' => $all->get_order()->filter( fn( $item ) => $item !== $id )->all(),
+		] );
+	}
+
+	public function patch( string $id, array $updated ) {
```
Comment (Nevoss) [inline modules/global-classes/repository.php:42]: What do we do, if the `$updated` is not valid data?

File: modules/global-classes/repository.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+		$this->kit->update_json_meta( self::META_KEY, [
+			'data' => $all->get_data()->merge( [ $id => $updated ] )->all(),
+			'order' => $all->get_order()->all(),
+		] );
+
+		return true;
+	}
+
+	public function create( array $class ) {
```
Comment (Nevoss) [inline modules/global-classes/repository.php:57]: same here

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: modules/global-classes/repository.php (hunk 1 above)
Comment (Omerisra6) [reply modules/global-classes/repository.php:42]: Shouldn't it be handled at the API part?

File: modules/atomic-widgets/styles/utils.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class Utils {
+	public static function generate_id( string $prefix = '', $existing_ids = [] ): string {
+		$existing_ids = array_flip( $existing_ids );
+
+		$id = $prefix . wp_rand( 1000000, 9999999 );
+
+		while ( isset( $existing_ids[ $id ] ) ) {
+			$id = $prefix . wp_rand( 1000000, 9999999 );
+		}
```
Comment (Nevoss) [inline modules/atomic-widgets/styles/utils.php:13]: use do while

```suggestion
		$id;

		do () {
			$id = $prefix . wp_rand( 1000000, 9999999 );
		} while ( in_array( $id, $existing_ids ) )
```

File: modules/global-classes/repository.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		}
+
+		return $this->all()->get_order()->all();
+	}
+
+	private function generate_global_class_id() {
+		$existing_ids = array_keys( $this->all()->get_items()->all() );
+		$kit_id = $this->kit->get_id();
+
+		return Atomic_Styles_Utils::generate_id( 'g-' . $kit_id . '-', $existing_ids );
```
Comment (Nevoss) [inline modules/global-classes/repository.php:103]:
```suggestion
		return Atomic_Styles_Utils::generate_id( 'g-' . $kit_id . '-', array_keys( $existing_ids ) );
```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: modules/global-classes/repository.php (hunk 4 above)
Comment (Omerisra6) [reply modules/global-classes/repository.php:101]: why?

File: modules/atomic-widgets/styles/utils.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Styles;
+
+class Utils {
+	public static function generate_id( string $prefix = '', $existing_ids = [] ): string {
+		$existing_ids = array_flip( $existing_ids );
```
Comment (Nevoss) [inline modules/atomic-widgets/styles/utils.php:7]: Why there is a need for flip? do you pass an array or ids? `['id-1', 'id-2', 'id-3']` if yes (it seems like you pass it the right way) there is no need to flip

**Review by Nevoss — CHANGES_REQUESTED** [review]

**Review by Nevoss — APPROVED** [review]

**Pull request merged**